src/webhooks/generic-notifier/generic-notifier.ts

```diff
@@ -0,0 +1,84 @@
+import { v1 } from '@datadog/datadog-api-client';
+import * as Sentry from '@sentry/node';
+import { FastifyReply, FastifyRequest } from 'fastify';
+import moment from 'moment-timezone';
+
+import { GenericEvent } from '@types';
+
+import { bolt } from '@/api/slack';
+import { DATADOG_API_INSTANCE } from '@/config';
+import { EVENT_NOTIFIER_SECRETS } from '@/config/secrets';
+import { extractAndVerifySignature } from '@/utils/auth/extractAndVerifySignature';
+
+export async function genericEventNotifier(
+  request: FastifyRequest<{ Body: GenericEvent }>,
+  reply: FastifyReply
+): Promise<void> {
+  try {
+    // If the webhook secret is not defined, throw an error
+    const { body }: { body: GenericEvent } = request;
+    if (
+      body.source === undefined ||
+      EVENT_NOTIFIER_SECRETS[body.source] === undefined
+    ) {
+      reply.code(400).send('Invalid source or missing secret');
+      throw new Error('Invalid source or missing secret');
+    }
+
+    const isVerified = await extractAndVerifySignature(
+      request,
+      reply,
+      'x-infra-hub-signature',
+      EVENT_NOTIFIER_SECRETS[body.source]
+    );
+    if (!isVerified) {
+      // If the signature is not verified, return (since extractAndVerifySignature sends the response)
+      return;
+    }
+
+    await messageSlack(body);
+    await sendEventToDatadog(body, moment().unix());
```

Comment:
The only other one was Jira, but never really got around to getting that added since Alex was (i think?) working a little on the env setup, so I was originally waiting for him to get that done first.